doc/design/service-discovery.md

```diff
@@ -172,9 +172,9 @@ Create a new publisher in RouDi which sends a `ServiceRegistryTopic`. This publi
 change in the service registry and to transmit the service discovery registry. The complete old service registry
 (saved locally) would be compared to the new service registry in a new class, extending the public user API.
 
-![overview diagram](diagrams/service-discovery/overview-alternative-b.svg)
+![overview diagram](diagrams/service-discovery/overview-alternative-d.svg)
```

Comment:
I am a little bit confused. The design document states 5 alternatives and argues in detail that alternative E is used.

And it states that alternative D can replace the interface port - which would be possible but an extreme performance hit (this is just a side note, please excuse my jibber jabber).

Furthermore alternative D states that a publisher is `used both to signal a change in the service registry and to transmit the service discovery registry`. 
Is this what you are currently implementing? If so then some parts seem to be missing. Could you please update the design document in a way that it is helpful for the review process and the reviewer knows what was/shall be implemented and why.

Reply:
This is correct. This PR implements `Alternative D` as an intermediate solution until #1096 is finished and the `StatusPort` is available. Adapted the design doc.


@FerdinandSpitzschnueffler Please update it on your PR with the changes you did.

Reply:
@mossmaurice If this PR is implementing alternative D you have to adjust alternative D in the design document since it implicitly talks about to publishers. One to signal a change and the other to transmit the service discovery registry.